MoreLinq/Rank.cs

```diff
@@ -45,7 +45,7 @@ public static IEnumerable<int> Rank<TSource>(this IEnumerable<TSource> source)
 
         public static IEnumerable<int> Rank<TSource>(this IEnumerable<TSource> source, IComparer<TSource> comparer)
         {
-            return source.RankBy(x => x, comparer);
+            return source.RankBy(Identity<TSource>, comparer);
```

Comment:
Type argument is redundant:

```suggestion
            return source.RankBy(Identity, comparer);
```